src/condor_credd/condor_credmon_oauth/condor_vault_storer

```diff
@@ -0,0 +1,202 @@
+#!/bin/bash
+#
+# Obtain and store condor credentials for each given oath_service 
+# Can be run automatically from condor_submit when configured as
+#   SEC_CREDENTIAL_STORER.
+
+ME="${0##*/}"
+
+ARGS="$@"
+
+# When condor_submit launches this the cursor is not at the beginning
+#   of a new line, so always send a newline before anything else to
+#   stdout or stderr.
+NEWLINENEEDED=true
+newlineifneeded()
+{
+    if $NEWLINENEEDED; then
+        echo
+        NEWLINENEEDED=false
+    fi
+}
+
+usage()
+{
+    newlineifneeded
+    echo "Usage: $ME [-vd] oauth_service ..."
+    echo "  -v shows more progress than normal" 
+    echo "  -d shows a lot of debug information" 
+    echo "  Each oauth_service is an issuer optionally followed by underscore and role"
+    echo "  Options may be added to each oauth_service:"
+    echo "    &handle=<name>       A name to be added to the credential"
+    echo "    &scope=<scopes>      A comma-separated list of scopes to request"
+    echo "    &audience=<audience> A token audience to request"
+    exit 1
+} >&2
+
+fatal()
+{
+    newlineifneeded
+    echo "$ME: $@"
+    if [ "$VERBOSE" = "" ]; then
+        echo "  More details may be available by running"
+        echo "    $ME -v $ARGS"
+    fi
+    exit 1
+} >&2
+
+verbose()
+{
+    if [ -n "$VERBOSE" ]; then
+        newlineifneeded
+        echo "$@" >&2
+    fi
+}
+
+VERBOSE=
+while getopts "dv" opt; do
+    case ${opt} in
+        d) VERBOSE="-d";;
+        v) VERBOSE="-v";;
+        \?) usage;;
+    esac
+done
+shift $((OPTIND -1))
+
+if [ "$#" = 0 ]; then
+    usage
+fi
+
+STOREOUT=/dev/null
+if [ "" != "$VERBOSE" ]; then
+    STOREOUT=/dev/stdout
+fi
+
+CONDOROPTS="$(condor_config_val SEC_CREDENTIAL_GETTOKEN_OPTS 2>/dev/null)"
+if [ -z "$CONDOROPTS" ] && [ -z "$HTGETTOKENOPTS" ]; then
+    fatal 'Neither SEC_CREDENTIAL_GETTOKEN_OPTS condor value nor $HTGETTOKENS environment set'
+fi
```

Comment:
The docs cover `SEC_CREDENTIAL_GETTOKEN_OPTS`, though any thoughts on making it `-a $(FULL_HOSTNAME)` by default? Assuming in many cases that Vault is on the same machine as the submitter... maybe that's a bad assumption though.

Also suggestion:
```suggestion
CONDOROPTS="$(condor_config_val SEC_CREDENTIAL_GETTOKEN_OPTS 2>/dev/null)"
if [ -z "$CONDOROPTS" ] && [ -z "$HTGETTOKENOPTS" ]; then
    fatal 'Neither SEC_CREDENTIAL_GETTOKEN_OPTS condor value nor $HTGETTOKENOPTS environment set'
fi
```

Reply:
Normally vault should not be on the same machine as the submitter.  It should be a separate, secured machine (likely a VM) that only system administrators can log into.